### ISSUE TITLE
Semantic token for _countof macro specifically is incorrect

### ISSUE THREAD

**Issue by leidegre** [microsoft/vscode-cpptools#12408]
### Environment

- OS and Version: Windows 11 [Version 10.0.22631.3737]
- VS Code Version: 1.90.2
- C/C++ Extension Version: 1.20.5
- If using SSH remote, specify OS of remote machine: n/a

### Bug Summary and Steps to Reproduce

This looks correct but note the extra whitespace after the first left parenthesis.

![image](https://github.com/microsoft/vscode-cpptools/assets/63085/c916dd4e-fb60-4015-ba92-b1fb6b5c24e8)

In the video below you can see what happens as I just change the whitespace.

https://github.com/microsoft/vscode-cpptools/assets/63085/63108375-e1a6-4c78-bff7-d1098a57c190

![image](https://github.com/microsoft/vscode-cpptools/assets/63085/2e46d64a-924e-41c8-b4b7-6b98b8d54f0d)

![image](https://github.com/microsoft/vscode-cpptools/assets/63085/c6863982-cdc2-447d-b081-bdebeb8a35ad)

It's incorrectly somehow counting part of the macro arguments as the macro itself. This seems to happen everywhere the `_countof` macro is used except inside other macros.

For example, here it's correct

![image](https://github.com/microsoft/vscode-cpptools/assets/63085/b1d1d720-0313-4e3d-a6c2-1c083005f964)

But everywhere else it is not

![image](https://github.com/microsoft/vscode-cpptools/assets/63085/430a99c3-2ba0-40ea-855b-4560ca7f24d9)

### Configuration and Logs

```shell
{
  "version": 4, 
  "configurations": [
    {
      "name": "Win32 (Debug)",
      "includePath": ["third-party/include", "src"],
      "defines": ["_DEBUG", "_CRT_SECURE_NO_WARNINGS"],
      "compilerArgs": ["/W4", "/wd4100", "/WX"],
      "windowsSdkVersion": "10.0.22621.0",
      "compilerPath": "cl.exe",
      "cStandard": "c17",
      "cppStandard": "c++20",
      "intelliSenseMode": "windows-msvc-x64"
    }
  ]
}

I don't have a "C/C++: Log Diagnostics" command in VS Code.
```

**Comment by browntarik** [comment]
Could you provide the code snippet that is causing the issue?

Does this issue persist when all other extension except for C/C++ are disabled?

**Comment by leidegre** [comment]
If I try to goto the definition of the _countof macro I end up here, in vcruntime.h. `__crt_countof` happens to be the same length as the incorrectly colored tokens, so it is somehow mistaking `_countof` for `__crt_countof`

```cpp
#ifdef __cplusplus
    extern "C++"
    {
        template <typename _CountofType, size_t _SizeOfArray>
        char (*__countof_helper(_UNALIGNED _CountofType (&_Array)[_SizeOfArray]))[_SizeOfArray];

        #define __crt_countof(_Array) (sizeof(*__countof_helper(_Array)) + 0)
    }
#else
    #define __crt_countof(_Array) (sizeof(_Array) / sizeof(_Array[0]))
#endif
```

The actual definition of `_countof` is in stdlib.h and looks like this

```cpp
#ifndef _countof
    #define _countof __crt_countof
#endif
```

This happens will all other extensions disabled.

**Comment by Colengms** [comment]
Hi @leidegre I can repro this issue with 1.20.5, but it appears to be fixed in the latest prerelease (1.21.0).  Closing as fixed.

**Comment by leidegre** [comment]
@Colengms Looks good on my machine now after update. Thanks!
